### PR TITLE
#57 Can't retrieve attributes with uppercase letters

### PR DESCRIPTION
**Pull request by tmjee**
#57 Can't retrieve attributes with uppercase letters
- Fix attributes always being converted to lower case

### CODE REVIEW

**Comment by dmarkov** [comment]
@tmjee Thanks, let me find someone who can review this pull request

**Comment by dmarkov** [comment]
@pinaf please review this one

File: src/test/java/com/jcabi/dynamo/AttributesTest.java [hunk 1]
```diff
@@ -103,25 +103,28 @@ public void filtersOutUnnecessaryKeys() throws Exception {
      */
     @Test
     public void caseInsensitive() throws Exception {
```
Comment (pinaf) [inline src/test/java/com/jcabi/dynamo/AttributesTest.java]: @tmjee we should rename this method to `caseSensitive`.

File: src/test/java/com/jcabi/dynamo/AttributesTest.java [hunk 2]
```diff
@@ -103,25 +103,28 @@ public void filtersOutUnnecessaryKeys() throws Exception {
...
+        final String alphaone = "Alpha";
```
Comment (pinaf) [inline src/test/java/com/jcabi/dynamo/AttributesTest.java]: @tmjee let's rename these to `first`, `second` and `third`.

File: src/test/java/com/jcabi/dynamo/AttributesTest.java [hunk 3]
```diff
@@ -103,25 +103,28 @@ public void filtersOutUnnecessaryKeys() throws Exception {
...
+                .with(alphatwo, "val-2"),
+            Matchers.hasKey(alphaone)
         );
         MatcherAssert.assertThat(
-            new Attributes()
-                .with("Beta", "some text to use")
-                .only(Collections.singletonList("bEta")),
-            Matchers.hasKey("beta")
+                new Attributes()
+                        .with(betaone, "some text to use")
```
Comment (pinaf) [inline src/test/java/com/jcabi/dynamo/AttributesTest.java]: @tmjee indentation was broken here

File: src/test/java/com/jcabi/dynamo/AttributesTest.java [hunk 4]
```diff
@@ -103,25 +103,28 @@ public void filtersOutUnnecessaryKeys() throws Exception {
...
+            Matchers.hasSize(2)
         );
         MatcherAssert.assertThat(
             new Attributes()
-                .with("Alpha", "val-1")
-                .with("AlPha", "val-2"),
-            Matchers.hasKey("alpha")
+                .with(alphaone, "val-1")
+                .with(alphatwo, "val-2"),
+            Matchers.hasKey(alphaone)
```
Comment (pinaf) [inline src/test/java/com/jcabi/dynamo/AttributesTest.java]: @tmjee let's check for both variables with `Matchers.allOf`

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java [hunk 5]
```diff
@@ -69,7 +69,7 @@ public void storesAndReadsAttributes() throws Exception {
         final String table = "users";
         final String key = "id";
         final int number = 43;
-        final String attr = "description";
+        final String attr = "DESCRIPTION";
```
Comment (pinaf) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: @tmjee why change this?

**Comment by pinaf** [comment]
@tmjee please see 5 comments above

**Comment by pinaf** [comment]
@tmjee I don't see why some tests were changed just to make the variables uppercase.

File: src/test/java/com/jcabi/dynamo/mock/H2DataTest.java (hunk 5 above)
Comment (tmjee) [inline src/test/java/com/jcabi/dynamo/mock/H2DataTest.java]: It's using H2 database and the columns from meta data are by default uppercase.

**Comment by tmjee** [comment]
@pinaf H2Data is using H2 database and by default meta data are in upper case. See http://www.h2database.com/javadoc/org/h2/constant/DbSettings.html for more info

**Comment by tmjee** [comment]
@pinaf  Fixed all above mentioned issues apart from variables uppercase.

**Comment by pinaf** [comment]
@tmjee alright

**Comment by pinaf** [comment]
@rultor merge

**Comment by pinaf** [comment]
@tmjee thanks!

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by dmarkov** [comment]
@pinaf Many thanks! **20 mins** were added to your account in Transaction ID `AP-3SF52554M42726428` (task took 96 hours and 30 mins). review comments (c=5) added as a bonus. +20 to your rating, your total score is [+4237](http://www.netbout.com/b/35128?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy now pls

**Comment by tmjee** [comment]
@dmarkov Can we close this ticket. Thx.

**Comment by pinaf** [comment]
@tmjee the PR is closed. ask the person who opened the original ticket to close it.

**Comment by dmarkov** [comment]
> @dmarkov Can we close this ticket. Thx.

@tmjee it is closed/merged already... or I didn't understand you?

**Comment by tmjee** [comment]
@dmarkov If we go to @dmarkov  If I go to https://github.com/jcabi/jcabi-dynamo/issues/57  it still says #57 is open. However the pull to fix it #59 is merged into master. So i think we should close issue #57. Did I understand the process correctly?

**Comment by pinaf** [comment]
@tmjee ask @nan-zhong to close it there since he's the one who opened it.

**Comment by dmarkov** [comment]
> @dmarkov If we go to https://github.com/jcabi/jcabi-dynamo/issues/57  it still says #57 is open. However the pull to fix it #59 is merged into master. So i think we should close issue #57. Did I understand the process correctly?

@tmjee you should ask @nan-zhong in #57 to close it, since that's who is the author of the problem

**Pull request merged**